drivers/dma/dma_mcux_edma.c

```diff
@@ -396,12 +401,26 @@ static int dma_mcux_edma_get_status(const struct device *dev,
 	return 0;
 }
 
+static bool edma_channel_filter(const struct device *dev,
+				int channel_id, void *param)
+{
+	enum dma_channel_filter *filter = (enum dma_channel_filter *)param;
+
+	if (filter && *filter == DMA_CHANNEL_PERIODIC) {
+		if (channel_id > 3) {
+			return false;
+		}
+	}
+	return true;
+}
+
 static const struct dma_driver_api dma_mcux_edma_api = {
 	.reload = dma_mcux_edma_reload,
 	.config = dma_mcux_edma_configure,
 	.start = dma_mcux_edma_start,
 	.stop = dma_mcux_edma_stop,
 	.get_status = dma_mcux_edma_get_status,
+	.chan_filter = edma_channel_filter,
```

Comment:
Can you keep this consistent with the `dma_mcux_*` nomenclature?

Reply:
ok